api-compose-object.go

```diff
@@ -408,12 +408,15 @@ func (c Client) uploadPartCopy(ctx context.Context, bucket, object, uploadID str
 	return p, nil
 }
 
-// ComposeObjectWithProgress - creates an object using server-side copying
+// ComposeObjectOptions holds options during compose objects
+type ComposeObjectOptions = CopyObjectOptions
+
+// ComposeObject - creates an object using server-side copying
 // of existing objects. It takes a list of source objects (with optional offsets)
 // and concatenates them into a new object using only server-side copying
 // operations. Optionally takes progress reader hook for applications to
 // look at current progress.
-func (c Client) ComposeObjectWithProgress(ctx context.Context, dst DestinationInfo, srcs []SourceInfo, progress io.Reader) error {
+func (c Client) ComposeObject(ctx context.Context, dst DestinationInfo, srcs []SourceInfo, opts ComposeObjectOptions) error {
```

Comment:
@vadmeste I am still not able to understand the need to `dst, srcs` can this be not generated inside? this function?